lib/ClientSai.cpp

```diff
@@ -1549,3 +1549,23 @@ sai_status_t ClientSai::queryApiVersion(
 
     return SAI_STATUS_NOT_IMPLEMENTED;
 }
+
+sai_status_t ClientSai::dbgGenerateDump(
+        _In_ const char *dump_file_name)
+{
+    MUTEX();
```

Comment:
Why do we need this MUTEX?

Reply:
I am not sure for the use case of using ClientSai ClientServerSai but i just based on other similar usecase such as 
sai_status_t ClientSai::queryAttributeCapability(
        _In_ sai_object_id_t switchId,
        _In_ sai_object_type_t objectType,
        _In_ sai_attr_id_t attrId,
        _Out_ sai_attr_capability_t *capability)
{
    MUTEX();
    SWSS_LOG_ENTER();
    REDIS_CHECK_API_INITIALIZED();

Reply:
the mutex is probably to protect that the replay will in the sender order